test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,406 @@
+         * collections. Each prefix is randomly generated. The result of the seach_near call with
+         * prefix enabled is then validated using the search_near call without prefix enabled.
+         */
+        logger::log_msg(
+          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
+
+        std::map<uint64_t, scoped_cursor> cursors;
+        const char *key_prefix, *key_default;
+        int exact_prefix, exact_default;
+        int ret;
+
+        while (tc->running()) {
+            /* Get a random collection to work on. */
+            collection &coll = tc->db.get_random_collection();
+
+            /* Find a cached cursor or create one if none exists. */
+            if (cursors.find(coll.id) == cursors.end()) {
+                cursors.emplace(
+                  coll.id, std::move(tc->session.open_scoped_cursor(coll.name.c_str())));
+                auto &cursor_prefix = cursors[coll.id];
+                /* The cached cursors have the prefix configuration enabled. */
+                testutil_check(
+                  cursor_prefix.get()->reconfigure(cursor_prefix.get(), "prefix_key=true"));
+            }
+
+            auto &cursor_prefix = cursors[coll.id];
+
+            /*
+             * Select a random timestamp between the oldest and now and start the transaction at
+             * that time. Get rid of the last 32 bits as they represent an increment for uniqueness.
+             */
+            wt_timestamp_t ts = random_generator::instance().generate_integer(
+              (tc->tsm->get_oldest_ts() >> 32), (tc->tsm->get_next_ts() >> 32));
+            /* Put back the timestamp in the correct format. */
+            ts <<= 32;
+            tc->transaction.begin("read_timestamp=" + tc->tsm->decimal_to_hex(ts));
+
+            /*
+             * The oldest timestamp might move ahead and the reading timestamp might become invalid.
+             * If this happens, we can exit the current loop.
+             */
+            while (tc->transaction.active() && tc->running() && ts >= tc->tsm->get_oldest_ts()) {
+
+                /*
+                 * Generate a random prefix. For this, we start by generating a random size and then
+                 * its value.
+                 */
+                const uint64_t prefix_size = random_generator::instance().generate_integer(
+                  static_cast<int64_t>(1), tc->key_size);
+                const std::string prefix = random_generator::instance().generate_random_string(
+                  prefix_size, characters_type::ALPHABET);
+
+                /* Open a cursor with the default configuration on the selected collection. */
+                scoped_cursor cursor_default(tc->session.open_scoped_cursor(coll.name.c_str()));
+
+                /* Call the search_near api using the two cursors. */
+                search_near(cursor_default, cursor_prefix, prefix);
```

Comment:
Suggestion: call search near with the prefix cursor and pass in the ret, cmp and key to search_near that can be renamed to `validate_search_near`